### ISSUE TITLE
Process crashes with "free(): invalid pointer"

### ISSUE THREAD

**Issue by RodrigoLeiteF** [mariusor/mpris-scrobbler#70]
Here's the output from `systemctl --user status mpris-scrobbler.service`

```
 % systemctl --user status mpris-scrobbler.service                                                  
● mpris-scrobbler.service - daemon to scrobble tracks loaded from the MPRIS DBus interface to compatible services
   Loaded: loaded (/usr/lib/systemd/user/mpris-scrobbler.service; enabled; vendor preset: enabled)
   Active: failed (Result: core-dump) since Wed 2019-10-23 12:48:33 -03; 3h 15min ago
     Docs: man:mpris-scrobbler
  Process: 4052 ExecStart=/usr/bin/mpris-scrobbler -vv (code=dumped, signal=ABRT)
 Main PID: 4052 (code=dumped, signal=ABRT)

out 23 12:48:33 rodrigo-laptop mpris-scrobbler[4052]: WARNING scrobble::valid::album[TURN OFF THE LIGHT]: yes
out 23 12:48:33 rodrigo-laptop mpris-scrobbler[4052]: WARNING scrobble::valid::length[236]: yes
out 23 12:48:33 rodrigo-laptop mpris-scrobbler[4052]: WARNING scrobble::valid::play_time[7.000000:118.000000]: no
out 23 12:48:33 rodrigo-laptop mpris-scrobbler[4052]: [48B blob data]
out 23 12:48:33 rodrigo-laptop mpris-scrobbler[4052]: WARNING scrobble::valid::scrobbled: yes
out 23 12:48:33 rodrigo-laptop mpris-scrobbler[4052]: [105B blob data]
out 23 12:48:33 rodrigo-laptop mpris-scrobbler[4052]: free(): invalid pointer
out 23 12:48:33 rodrigo-laptop systemd[4046]: mpris-scrobbler.service: Main process exited, code=dumped, status=6/ABRT
out 23 12:48:33 rodrigo-laptop systemd[4046]: mpris-scrobbler.service: Failed with result 'core-dump'.
out 23 12:48:34 rodrigo-laptop systemd-coredump[7590]: Process 4052 (mpris-scrobbler) of user 1000 dumped core.

                                                       Stack trace of thread 4052:
                                                       #0  0x00007f7ee7ce6f25 raise (libc.so.6)
                                                       #1  0x00007f7ee7cd0897 abort (libc.so.6)
                                                       #2  0x00007f7ee7d2a258 __libc_message (libc.so.6)
                                                       #3  0x00007f7ee7d3177a malloc_printerr (libc.so.6)
                                                       #4  0x00007f7ee7d3314c _int_free (libc.so.6)
                                                       #5  0x0000555a453d136c n/a (mpris-scrobbler)
                                                       #6  0x0000555a453d1b87 n/a (mpris-scrobbler)
                                                       #7  0x0000555a453d2729 n/a (mpris-scrobbler)
                                                       #8  0x00007f7ee7ea6430 n/a (libevent-2.1.so.7)
```

```
 % coredumpctl -o mpris-scrobbler.dump dump 4052                                                    Repos/upnid-cli (master) rodrigo-laptop
           PID: 4052 (mpris-scrobbler)
           UID: 1000 (rodrigo)
           GID: 1000 (rodrigo)
        Signal: 6 (ABRT)
     Timestamp: Wed 2019-10-23 12:48:33 -03 (3h 31min ago)
  Command Line: /usr/bin/mpris-scrobbler -vv
    Executable: /usr/bin/mpris-scrobbler
 Control Group: /user.slice/user-1000.slice/user@1000.service/mpris-scrobbler.service
          Unit: user@1000.service
     User Unit: mpris-scrobbler.service
         Slice: user-1000.slice
     Owner UID: 1000 (rodrigo)
       Boot ID: 14cb75c77eca414bad71bad7521f9416
    Machine ID: 87aa987fe0a1408c9c87508ab3403a78
      Hostname: rodrigo-laptop
       Storage: none
       Message: Process 4052 (mpris-scrobbler) of user 1000 dumped core.

                Stack trace of thread 4052:
                #0  0x00007f7ee7ce6f25 raise (libc.so.6)
                #1  0x00007f7ee7cd0897 abort (libc.so.6)
                #2  0x00007f7ee7d2a258 __libc_message (libc.so.6)
                #3  0x00007f7ee7d3177a malloc_printerr (libc.so.6)
                #4  0x00007f7ee7d3314c _int_free (libc.so.6)
                #5  0x0000555a453d136c n/a (mpris-scrobbler)
                #6  0x0000555a453d1b87 n/a (mpris-scrobbler)
                #7  0x0000555a453d2729 n/a (mpris-scrobbler)
                #8  0x00007f7ee7ea6430 n/a (libevent-2.1.so.7)
                #9  0x00007f7ee7ea6b7f event_base_loop (libevent-2.1.so.7)
                #10 0x0000555a453c70f5 n/a (mpris-scrobbler)
                #11 0x00007f7ee7cd2153 __libc_start_main (libc.so.6)
                #12 0x0000555a453c71ae n/a (mpris-scrobbler)
Coredump entry has no core attached (neither internally in the journal nor externally on disk).
```

Using ncmpcpp on Manjaro Linux.

**Comment by mariusor** [comment]
Thank you for taking the time to submit this. I'm getting them too on my machine, so it's sort-of a known issue, probably due to iffy string values loading from dbus.

However I don't have too much time to dedicate to the scrobbler lately, as my focus is on another project. I'll get to it sometime, but I don't know when.

**Comment by RodrigoLeiteF** [comment]
No problem. Take your time :)

**Comment by mariusor** [comment]
This should be fixed in the latest v0.3.92 and in the future release v0.4.
